src/lib/api/ext/img-srcset.spec.ts

```diff
@@ -0,0 +1,353 @@
+    expect(nodes[0].nativeElement).toHaveAttributes({
+      srcset: `${SRCSET_URLS_MAP['lt-xl'][0]}`,
+      media: breakpoints.findByAlias('lt-xl').mediaQuery
+    });
+    expect(nodes[1].nativeElement).toHaveAttributes({
+      srcset: `${SRCSET_URLS_MAP['lt-lg'][0]}`,
+      media: breakpoints.findByAlias('lt-lg').mediaQuery
+    });
+    expect(nodes[2].nativeElement).toHaveAttributes({
+      srcset: `${SRCSET_URLS_MAP['xs'][0]}`,
+      media: breakpoints.findByAlias('xs').mediaQuery
+    });
+  });
+
+  describe('with responsive api', () => {
+
+    it('should work with a isolated image element and responsive srcset(s)', () => {
+      componentWithTemplate(`
+        <img [srcset]="xsSrcSet"
+             [srcset.md]="mdSrcSet">
+      `);
+
+      let img = queryFor(fixture, 'img')[0].nativeElement;
+
+      matchMedia.activate('md');
+      fixture.detectChanges();
+      expect(img).toBeDefined();
+      expect(img).toHaveAttributes({
+        src: fixture.componentInstance.mdSrcSet
+      });
+
+      // When activating an unused breakpoint, fallback to default [srcset] value
+      matchMedia.activate('xl');
+      fixture.detectChanges();
+      expect(img).toHaveAttributes({
+        src: fixture.componentInstance.xsSrcSet
+      });
+    });
+
+    it('should work use [src] if default [srcset] is not defined', () => {
+      componentWithTemplate(`
+         <img src="${DEFAULT_SRC}"
+              [srcset.md]="mdSrcSet">
+       `);
+      fixture.detectChanges();
+      let img = queryFor(fixture, 'img')[0].nativeElement;
+
+      matchMedia.activate('md');
+      fixture.detectChanges();
+
+      expect(img).toBeDefined();
+      expect(img).toHaveAttributes({
+        src: fixture.componentInstance.mdSrcSet
+      });
+
+      // When activating an unused breakpoint, fallback to default [srcset] value
+      matchMedia.activate('xl');
+      fixture.detectChanges();
+      expect(img).toHaveAttributes({
+        src: DEFAULT_SRC
+      });
+    });
+
+    it('should work with lookup from `src` when the default `srcset` is not defined', () => {
+      componentWithTemplate(`
+        <picture>
+            <img  src="https://dummyimage.com/400x200/c7c224/000.png&text=default"
+                  srcset.md="https://dummyimage.com/500x200/76c720/fff.png&text=md">
+        </picture>
+      `);
+      matchMedia.activate('md');
+      fixture.detectChanges();
+
+      let sources = queryFor(fixture, 'source');
+      let defaultSourceEl = sources[sources.length - 1].nativeElement;
+      expect(defaultSourceEl).toHaveAttributes({
+        srcset: 'https://dummyimage.com/500x200/76c720/fff.png&text=md'
+      });
+
+      // When activating an unused breakpoint, fallback to default [srcset] value
+      matchMedia.activate('xs');
+      fixture.detectChanges();
+      expect(defaultSourceEl).toHaveAttributes({
+        srcset: 'https://dummyimage.com/400x200/c7c224/000.png&text=default'
```

Comment:
Shouldn't the source element have the srcset value 'https://dummyimage.com/500x200/76c720/fff.png&text=md' (and the media query corresponding to md) ?

When 'xs' is activated, the injected source element should not be updated. The browser will default to the img src value.